samples/build-duration-logger/index.html

```diff
@@ -139,47 +147,116 @@
-                    buildEventHandlers[eventType].forEach(handler => handler[`on${eventType}`](buildEventPayload));
-                }
-            });
-
-            // there isn't an onclose event that we can listen to, but an error event does get triggered
-            // when the end of the stream is reached so use that as a rough proxy for the end of the stream
-            buildEventStream.onerror = event => {
-                if (buildEventStreamOpen) {
+            const buildEventStreamUrl = this.createBuildEventStreamUrl(build.buildId);
+            const buildStream = createServerSideEventStream(buildEventStreamUrl, {
+                oncomplete:  () => {
+                    // there isn't an onclose event that we can listen to, but an error event does get triggered
+                    // when the end of the stream is reached so use that as a rough proxy for the end of the stream
                     this.finishedProcessingBuild();
 
                     // Call the 'complete()' method on any handler that has it.
                     if (buildEventHandlers.complete) {
                         buildEventHandlers.complete.forEach(handler => handler.complete());
                     }
+                },
+                eventListeners: [
+                    {
+                        eventName: 'BuildEvent',
+                        eventHandler:event => {
+                            const buildEventPayload = JSON.parse(event.data);
+                            const { eventType } = buildEventPayload.type;
 
-                    buildEventStream.close();
-                    buildEventStreamOpen = false;
+                            if (this.allHandledEventTypes.has(eventType)) {
+                                buildEventHandlers[eventType].forEach(handler => handler[`on${eventType}`](buildEventPayload));
+                            }
+                        }
+                    }
+                ],
+                retry: {
+                    interval: 2000,
+                    maxRetries: 100
                 }
-            }
+            });
         }
 
         finishedProcessingBuild() {
             this.numBuildsInProcess--;
             setTimeout(() => this.processPendingBuilds(), 0); // process the next set of pending builds, if any
         }
-    }; 
-        
-    new BuildProcessor(
-      GRADLE_ENTERPRISE_SERVER_URL, 
-      MAX_CONCURRENT_BUILDS_TO_PROCESS, 
-      BUILD_EVENT_HANDLERS
-    ).start(PROCESS_FROM);
-    
+    }
+
+    function createServerSideEventStream(url, configuration) {
+        const STATUS_COMPLETE = 204;
+
+        let stream;
+        let retries;
+
+        const _onopen = configuration.onopen || noop;
+        const _onerror = configuration.onerror || noop;
+        const _oncomplete = configuration.oncomplete || noop;
+        const _configurationRetry = configuration.retry || { }
+        const _maxRetries = _configurationRetry.maxRetries || 0;
```

Comment:
I'd use the following default to have a sane value and to not disable the retry functionality at all by default.
```suggestion
        const _maxRetries = _configurationRetry.maxRetries || 3;
```